### ISSUE TITLE
Blurry video

### ISSUE THREAD

**Issue by Wuerfel21** [EasyRPG/Player#953]
**Name of the game**: Any

**Player platform** (e.g. Windows, Linux, Android, ...): Wii (actually, Dolphin 5.0, don't have a Wii 'round currently, but I verified that the issue does not occur in other games, so I assume this is correctly emulated.)

**Attach files** (as a .zip archive or link them):
[blurry-rpg.zip](https://github.com/EasyRPG/Player/files/384120/blurry-rpg.zip)

**Describe the issue in detail and how to reproduce it**:
On the Wii, all graphics are significantly blurred, probably as a result of rendering the finished frame to the XFB(or was it EFB?) with (bi?)linear filtering. This occurs both on the current version and on 0.4.0 (which was before the overscan fix). A different game (or in this case, the 240p test suite in 480i mode) is pixel-perfect (same emulator settings). A texture dump is also pixel-perfect.

Proposed solution: Make overscan configurable (some TVs don't have significant overscan/can be adjusted.), seperately for horizontal and vertical(some PAL TV's "squish" 60Hz video vertically => less/no overscan). If both are off (full frame used), then use nearest neighbor scaling.
Additionally, a custom video mode like 336@240p might be used to reduce the width of the image without scaling (not sure if that would work, [take a look at SNES9x GX's "original" mode for implementation](https://github.com/dborth/snes9xgx/blob/609746a183a60930114ce05406b758ac53b3c1e1/source/video.cpp#L188)). It would also reduce flicker on CRT's...

**Comment by carstene1ns** [comment]
Yeah, the scaling problem is a known issue. I want to have a libwiigui-powered gui with configuration options somewhere in the future, to be able to adjust this as user.
Also, to implement that "proper" 240p output, we need to patch big hunks of the rendering part of SDL-Wii or write our own renderering code for Wii (i.e. ditch SDL completely), which is also on the longterm list, because we want to drop SDL1.2 support one day anyway.
However, before working on this, we need to have proper audio code for sound effects on platforms without SDL (which is the only thing currently hindering the PSP port btw.).

I am tagging this as a 0.6. version issue, but no guarantees.

**Comment by Wuerfel21** [comment]
Well, blitting a bitmap to the screen doesn't seem hard, i might have a go at it in a week or so (provided I manage to disable the Video portion of SDL without breaking the audio...)

**Comment by carstene1ns** [comment]
Yes, it is not hard. Rather not a high priority issue.

**Comment by Ghabry** [comment]
We assume our user count for the Wii version is less then 10 users. You are actually lucky to have a usable version by now because we needed 3 years to fix the audio issues on the Wii version ;)

**Comment by gadesx** [comment]
320x240p directly in CRT. I want to see that

**Comment by fdelapena** [comment]
This seems fixed since #3332
